compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/input/pointer/PointerEvent.skiko.kt

```diff
@@ -128,42 +123,114 @@ actual data class PointerEvent internal constructor(
      */
     val nativeEvent: Any?,
 
-    internal val _button: PointerButton?
+    button: PointerButton?
 ) {
     internal actual constructor(
         changes: List<PointerInputChange>,
         internalPointerEvent: InternalPointerEvent?
     ) : this(
-        changes,
-        internalPointerEvent?.buttons ?: PointerButtons(0),
-        internalPointerEvent?.keyboardModifiers ?: PointerKeyboardModifiers(0),
-        internalPointerEvent?.type ?: PointerEventType.Unknown,
-        internalPointerEvent?.nativeEvent,
-        internalPointerEvent?.button
+        changes = changes,
+        buttons = internalPointerEvent?.buttons ?: PointerButtons(0),
+        keyboardModifiers = internalPointerEvent?.keyboardModifiers ?: PointerKeyboardModifiers(0),
+        type = internalPointerEvent?.type ?: PointerEventType.Unknown,
+        nativeEvent = internalPointerEvent?.nativeEvent,
+        button = internalPointerEvent?.button
     )
 
     /**
      * @param changes The changes.
      */
     actual constructor(changes: List<PointerInputChange>) : this(
-        changes,
+        changes = changes,
         buttons = PointerButtons(0),
         keyboardModifiers = PointerKeyboardModifiers(0),
-        _type = PointerEventType.Unknown,
+        type = PointerEventType.Unknown,
         nativeEvent = null,
-        _button = null
+        button = null
     )
 
     /**
      * Specifies the pointer button state of which was changed.
      * It has value only when event's type is [PointerEventType.Press] or [PointerEventType.Release] and
      * when button is applicable (only for Mouse and Stylus events).
      */
+    @Suppress("CanBePrimaryConstructorProperty")
     @ExperimentalComposeUiApi
-    val button: PointerButton? = _button
+    val button: PointerButton? = button
 
-    actual var type: PointerEventType = _type
+    actual var type: PointerEventType = type
         internal set
+
+
+    // Backwards compatibility for when PointerEvent was a data class
+
+    fun component1(): List<PointerInputChange> = changes
+
+    fun component2(): PointerButtons = buttons
+
+    fun component3(): PointerKeyboardModifiers = keyboardModifiers
+
+    // _type was internal, so no need for a component4
```

Comment:
We still need public component4() for binary compatibility.

Also, user code may be like the following:
```val (a, b, c, d, e) = pointerEvent``` ( It's also using `component4()` )

Reply:
Or, because of internal, component4 was also internal?

Reply:
Sorry, I am wrong. We don't need component4.